daml/Tests/BinaryOption.daml

```diff
@@ -25,13 +28,14 @@ endToEnd = script do
                        signatories = Set.singleton providers.bank
                        label = "TSLA"
                        version = 0
-      underlyingAsset = Asset with id = underlyingId, quantity = 200.0
+      strike = 910.0: Decimal
+      expiration = date 2020 Mar 18
 
   alice <- onboardCustomer providers "Alice"
   bob <- onboardCustomer providers "Bob"
 
   -- Alice originates the option
-  let boClaims = binaryCallOption underlyingId usdId 700.0 (date 2022 Oct 22)
+  let boClaims = binaryCallOption underlyingId usdId strike expiration
```

Comment:
I believe this was unintentional, hopefully fixed now.

Reply:
In both cases this was intentional.

The price is different because the payoff/cashflow of this instrument during settlement is the `strike - price`, so if they were the same you would settle 0$.

The date (not just the month) was different in order to avoid triggering the lifecycle. This was because there was a bug in the lifecycle process at the time, so I wanted to avoid it triggering. I guess I just forgot to put it back to the expiry date. It should be safe to do now.

Reply:
Isn't the price 920 at expiry (see line 75)?
The assetLabel (line 40) has the strike, doesn't it?

Reply:
> Was this unintentionally different (700 vs 910, Oct vs Mar)?
Yes, I see what you mean now. I thought you were comparing the prices and expiry to that of the strike/ledger date. I didn't realise you meant the `assetLabel`. So rather yes, the `assetLabel` is incorrect.

Ideally we should just generate the `assetLabel` programmatically below to ensure it matches the option parameters.